sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -127,6 +127,49 @@ gnrc_pktsnip_t *_dio_dodag_conf_build(gnrc_pktsnip_t *pkt, gnrc_rpl_dodag_t *dod
     return opt_snip;
 }
 
+gnrc_pktsnip_t *_dis_solicited_opt_build(gnrc_pktsnip_t *pkt, gnrc_rpl_instance_t *inst)
+{
+    gnrc_pktsnip_t *opt_snip;
+    size_t snip_size = 0;
+#ifndef GNRC_RPL_DIS_WITHOUT_SOL
+    snip_size = sizeof(gnrc_rpl_opt_dis_solicited_t);
+#else
+    /* The DIS is too small so that wireshark complains about an incorrect
+     * ethernet frame check sequence.
+     * To trick it we PadN 2 additional bytes, i.e. 4 bytes in sum. */
+    uint8_t padding[] = {
+            GNRC_RPL_OPT_PADN,  /* Option Type */
+            0x02,               /* Number of extra padding bytes */
+            0x00, 0x00
+    };
+
+    snip_size = sizeof(padding);
+#endif
+
+    if ((opt_snip = gnrc_pktbuf_add(pkt, NULL, snip_size,
+                                    GNRC_NETTYPE_UNDEF)) == NULL) {
+        DEBUG("RPL: BUILD SOLICITED OPT - no space left in packet buffer\n");
+        gnrc_pktbuf_release(pkt);
+        return NULL;
+    }
+#ifndef GNRC_RPL_DIS_WITHOUT_SOL
+    gnrc_rpl_opt_dis_solicited_t* solicited_information;
+    solicited_information = opt_snip->data;
+
+    solicited_information->type = GNRC_RPL_OPT_SOLICITED_INFO;
+    solicited_information->length = 19;
```

Comment:
This _magic number_ should be replaced either with a define macro or some functional call? And provide some rational (docu) on the maths why its 19 and not 42?